iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/DuplicatedEnvironmentVariablesCheck.java

```diff
@@ -96,17 +101,19 @@ private void checkEnvironmentVariableFrom(BlockObject root, BlockObject envFrom,
    * @param creator A function used to wrap the resulting {@link YamlTree} into the corresponding {@link AbstractVariable} class instance.
    */
   private void checkMapResource(BlockObject root, BlockObject mapRef, Container container, Class<? extends MapResource> clazz,
-    Function<YamlTree, ? extends AbstractVariable> creator) {
+    TriFunction<YamlTree, String, YamlTree, ? extends AbstractVariable> creator) {
     retrieveMapRefTree(mapRef).ifPresent((ScalarTree mapRefNameValueTree) -> {
       var mapName = mapRefNameValueTree.value();
       getGlobalResources(root).stream()
         .filter(clazz::isInstance)
         .map(clazz::cast)
         .filter(mapResource -> mapName.equals(mapResource.name()))
-        .flatMap(mapResource -> mapResource.values().entrySet().stream())
-        .forEach((Map.Entry<String, TupleTree> entryConfigMap) -> {
-          var name = entryConfigMap.getKey();
-          container.variables.computeIfAbsent(name, key -> new ArrayList<>()).add(creator.apply(mapRefNameValueTree));
+        .forEach((MapResource mapResource) -> {
+          var filePath = mapResource.filePath();
+          mapResource.values().forEach((String name, TupleTree value) -> {
+            var variableKeyReferenceInMapResourceFile = value.key();
+            container.variables.computeIfAbsent(name, key -> new ArrayList<>()).add(creator.apply(mapRefNameValueTree, filePath, variableKeyReferenceInMapResourceFile));
```

Comment:
This should be in separate lines for clarity:
```suggestion
            container.variables
            .computeIfAbsent(name, key -> new ArrayList<>())
            .add(creator.apply(mapRefNameValueTree, filePath, variableKeyReferenceInMapResourceFile));
```